contents/docs/plugins/build/index.md

```diff
@@ -4,37 +4,53 @@ sidebar: Docs
 showTitle: true
 ---
 
-A great way to understand what you can do with [plugins](/docs/plugins/overview) is to understand how data flows throw plugins.
+PostHog makes it possible to build your own [plugins](/docs/plugins/overview) and integrate with other platforms. So, if [our integration library](/integrations) is missing something you need then you may still be able to create it yourself.
 
-There are 2 big concepts to remember:
+Plugins can add more information to the event, modify existing properties or trigger a range of other activities. There are also some plugins that enqueue jobs to run in the future. Find out more about jobs in [our developer reference docs](/docs/plugins/build/reference#jobs-1).
 
-1. Every plugin acts on a single event coming in.
+Before building your first plugin it's important to understand how data flows through plugins in the first place. There are three critical concepts to remember:
 
-2. Plugins act on events before they are stored.
+1. Plugins act on _single events_ coming in to PostHog.
 
-Each plugin can add more information to the event, or even modify existing properties. For example, the [GeoIP plugin](https://posthog.com/plugins/geoip) is a plugin that adds GeoIP information to the event.
+2. Plugins act on events _before_ they are stored.
```

Comment:
(looks like yes given the migrator plugin)

Reply:
iiuc there's both 0 before (e.g. geoIP) or after (e.g. export to bigquery).

Reply:
Hmm, the plugin acts on the event before ingestion, even in case of export, it's just that the export happens asynchronously, which can be before or after ingestion finishes 🤷 . So I'd vote for not adding this extra detail here.

Reply:
When it comes to events, plugins can access them at multiple stages:

1. Pre-ingestion (`processEvent`)
2. Alongside ingestion (`onEvent` - it runs after `processEvent` and simultaneously with storing the event in the db via `ingestEvent`)
3. Historical events - plugins using `exportEvents` (this is `onEvent` with some sugar on top) get a job called `Export historical events` that pipes a stream of previously ingested events through the plugin